libraries/stdlib/stdlib_ng.mtlx

```diff
@@ -1160,6 +1160,143 @@
     <output name="out" type="float" nodename="N_range" />
   </nodegraph>
 
+  <!--
+    Node: <randomfloat>
+    Produces a randomized float, based on an 'input' signal and 'seed' value.
+  -->
+  <nodegraph name="NG_randomfloat_float" nodedef="ND_randomfloat_float">
+    <convert name="N_convertSeed1" type="float">
+      <input name="in" type="integer" interfacename="seed" />
+    </convert>
+    <modulo name="N_moduloInput" type="float">
+      <input name="in1" type="float" interfacename="in" />
+      <input name="in2" type="float" value="4219" />
+    </modulo>
+    <multiply name="N_scaleInput" type="float">
+      <input name="in1" type="float" nodename="N_moduloInput" />
+      <input name="in2" type="float" value="3947" />
+    </multiply>
+    <combine2 name="N_combine2" type="vector2">
+      <input name="in1" type="float" nodename="N_scaleInput" />
+      <input name="in2" type="float" nodename="N_convertSeed1" />
+    </combine2>
+    <cellnoise2d name="N_cellnoise1" type="float">
+      <input name="texcoord" type="vector2" nodename="N_combine2" />
+    </cellnoise2d>
+    <range name="N_remapRange" type="float">
+      <input name="in" type="float" nodename="N_cellnoise1" />
+      <input name="outlow" type="float" interfacename="min" />
+      <input name="outhigh" type="float" interfacename="max" />
+      <input name="doclamp" type="boolean" value="true" />
+    </range>
+    <output name="out" type="float" nodename="N_remapRange" />
+  </nodegraph>
+  <nodegraph name="NG_randomfloat_integer" nodedef="ND_randomfloat_integer">
+    <convert name="N_convertInput1" type="float">
+      <input name="in" type="integer" interfacename="in" />
+    </convert>
+    <convert name="N_convertSeed1" type="float">
```

Comment:
From this point onward, could the `integer` version of this node simply call down to the `float` version?  I do see some logic differences between them, e.g. the `modulo` operations that are skipped here, but it's not clear that these differences are intentional.

Reply:
Yes those differences are intentional. Originally integer did re-use float version. But after internal discussions, forcing the integer signature to convert to float looked like the wrong approach. Float inputs need the modulo/scale to minimize quantization. This isn't necessary for integer inputs, because whole numbers lend themselves to cellnoise naturally.

Reply:
Got it, and thanks for those clarifications!